snapcraft/internal/build_providers/_base_provider.py

```diff
@@ -271,13 +237,16 @@ def launch_instance(self) -> None:
             os.makedirs(self.provider_project_dir)
             # then launch
             self._launch()
-            # and do first boot related things and if any failure occurs,
-            # then clean up.
+
+            # Configure environment for snapcraft use.
             self._setup_environment()
-        else:
-            # We always setup snapcraft after a start to bring it up to speed with
-            # what is on the host
-            self._setup_snapcraft()
+
+            # Refresh repository caches.
+            self._run(["snapcraft", "refresh"])
```

Comment:
btw,  why not just apt-get update? Now I recall that "snapcraft refresh" is to be run outside the provider

Reply:
you do not have snapcraft installed at this point for this to work. I would just remove this logic here (the else), as apt update was just run and we don't want to be refreshing on every instance start.

Reply:
I wondered that myself.  But looking at it, I suppose it can be.  And we can probably remove refresh altogether now?